### PR TITLE
#27 error handling

### PR DESCRIPTION
**Pull request by andreas-haller**
- upgraded groovy console version to be AEM 6.2+ compatible
- general fixes of exception handling
- fixed exception handling for GroovyConsole scripts
- fixed content hash handling
- added session.save() after each successful action execution

### CODE REVIEW

File: vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/UpgradeProcessor.java [hunk 1]
```diff
@@ -55,46 +55,65 @@
...
+                case INSTALL_FAILED:
+                    executeActions(ctx);
+                    break;
+                case END:
+                    executeActionsAndSaveStatus(ctx);
+                    break;
             }
         } catch (Exception e) {
+            failed = true;
             LOG.error(ctx, "Error during content upgrade", e);
```
Comment (kwin) [inline vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/UpgradeProcessor.java]: Please log or throw, never both!

File: vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/handler/groovy/GroovyScript.java [hunk 2]
```diff
@@ -44,8 +44,13 @@ public void execute(final InstallContext ctx) throws RepositoryException {
         if (request != null) {
             LOG.debug(ctx, "Executing [{}]", getName());
             final RunScriptResponse scriptResponse = run(request);
-            LOG.info(ctx, "Executed [{}]: [{}]\n{}\n---\n", getName(), scriptResponse.getRunningTime(),
-                    scriptResponse.getOutput().trim());
+            if (scriptResponse.getExceptionStackTrace() != null && scriptResponse.getExceptionStackTrace().trim().length() > 0) {
+                LOG.info(ctx, "Exception executing script [{}]: [{}]\n{}\n---\n", getName(), scriptResponse.getRunningTime(),
```
Comment (kwin) [inline vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/handler/groovy/GroovyScript.java]: never log and throw. Either or

**Review by kwin — CHANGES_REQUESTED** [review]

**Comment by andreas-haller** [comment]
@kwin Thanks for the feedback, fixed the two remarks.

File: vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/handler/groovy/GroovyScript.java [hunk 3]
```diff
@@ -44,9 +44,9 @@ public void execute(final InstallContext ctx) throws RepositoryException {
         if (request != null) {
             LOG.debug(ctx, "Executing [{}]", getName());
             final RunScriptResponse scriptResponse = run(request);
+            LOG.debug(ctx, "Executed script [{}]: [{}]\n{}\n---\n", getName(), scriptResponse.getRunningTime(),
+                    scriptResponse.getExceptionStackTrace().trim());
```
Comment (kwin) [inline vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/handler/groovy/GroovyScript.java]: this might be null here!

**Review by kwin — CHANGES_REQUESTED** [review]

**Review by kwin — APPROVED** [review]

**Pull request merged**